lib/irb/cmd/ls.rb

```diff
@@ -39,8 +39,19 @@ def execute(*arg, grep: nil)
       def dump_methods(o, klass, obj)
         singleton_class = begin obj.singleton_class; rescue TypeError; nil end
         dumped_mods = Array.new
+        ancestors = klass.ancestors
+        singleton_ancestors = singleton_class&.ancestors || []
+
+        if klass < Object
+          ancestors = ancestors.reject { |c| c >= Object }
+        end
+
+        if singleton_class && singleton_class < Object
```

Comment:
I think this `if` is not needed because `singleton_class < Object` almost always returns true

Reply:
Good catch 👍 Updated